File: lessons/compiled_lang/java/src/i10.subroutine.java
class I10Subroutine {
    // static fields are directly visible and mutable from any static
    //  method in this class - no "global" keyword needed like Python.
    static int pond = 500;
    static int captured = 0;

    static void fish() {
        pond -= 150;
        captured += 150;
    }

    public static void main(String[] args) {
        System.out.println("We have " + pond + " in this pond.");

        fish();
        System.out.println("Fishing from the main pond... We now have " + pond + " in the main pond.");

        fish();
        System.out.println("Fishing from the main pond... We now have " + pond + " in the main pond.");

        fish();
        System.out.println("Fishing from the main pond... We now have " + pond + " in the main pond.");

        System.out.println("We now have a total of " + captured + " fish captured");
    }
}

File: lessons/compiled_lang/rust/src/i10.subroutine.rs
// mutable statics can only be touched inside `unsafe` - Rust's
//  compiler-enforced stand-in for Python's explicit `global` keyword.
static mut POND: i32 = 500;
static mut CAPTURED: i32 = 0;

fn fish() {
    unsafe {
        POND -= 150;
        CAPTURED += 150;
    }
}

fn main() {
    let pond = unsafe { POND };
    println!("We have {pond} in this pond.");

    fish();
    let pond = unsafe { POND };
    println!("Fishing from the main pond... We now have {pond} in the main pond.");

    fish();
    let pond = unsafe { POND };
    println!("Fishing from the main pond... We now have {pond} in the main pond.");

    fish();
    let pond = unsafe { POND };
    println!("Fishing from the main pond... We now have {pond} in the main pond.");

    let captured = unsafe { CAPTURED };
    println!("We now have a total of {captured} fish captured");
}
